internal/locate/region_cache.go

```diff
@@ -120,8 +122,10 @@ func SetRegionCacheTTLSec(t int64) {
 }
 
 const (
-	updated  int32 = iota // region is updated and no need to reload.
-	needSync              // need sync new region info.
+	needReloadOnAccess     int32 = 1 << iota // indicates the region will be reloaded on next access
+	needExpireAfterTTL                       // indicates the region will expire after RegionCacheTTL (even when it's accessed continuously)
```

Comment:
The strategy is to expire these regions with certain conditions like down peers, we may need to consider how to make the region reloads smoother as the current TTL is a constant value.

Reply:
I think it's relative fine for regions that have stale or unreachable peers, since the cache GC only scan 50 regions per seconds. For regions that have down peers, maybe we can add a random value to `lastAccess` when constructing them by `newRegion`?

Reply:
Maybe we could do some improvements about smoothing region reloads in another PR, considering both expire reloading and region cache miss reloading. I rembember we've encountered lots of region reload caused by sudden TTL expire or something like it.

Reply:
Agreed. One idea I'm thinking about is: we may change `lastAccess` to `TTL` and push it as `now + RegionCacheTTL + RandDev` on access.